homeassistant/components/netatmo/select.py

```diff
@@ -35,100 +26,66 @@ async def async_setup_entry(
-    ]
+    @callback
+    def _create_entity(netatmo_home: NetatmoHome) -> None:
+        entity = NetatmoScheduleSelect(netatmo_home)
+        async_add_entities([entity])
 
-    _LOGGER.debug("Adding climate schedule select entities %s", entities)
-    async_add_entities(entities, True)
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, NETATMO_CREATE_SELECT, _create_entity)
+    )
 
 
 class NetatmoScheduleSelect(NetatmoBase, SelectEntity):
     """Representation a Netatmo thermostat schedule selector."""
 
     def __init__(
-        self, data_handler: NetatmoDataHandler, home_id: str, options: list
+        self,
+        netatmo_home: NetatmoHome,
     ) -> None:
         """Initialize the select entity."""
         SelectEntity.__init__(self)
-        super().__init__(data_handler)
-
-        self._home_id = home_id
-
-        self._climate_state_class = f"{CLIMATE_STATE_CLASS_NAME}-{self._home_id}"
-        self._climate_state: pyatmo.AsyncClimate = data_handler.data[
-            self._climate_state_class
-        ]
+        super().__init__(netatmo_home.data_handler)
 
-        self._home = self._climate_state.homes[self._home_id]
+        self._home = netatmo_home.home
+        self._home_id = self._home.entity_id
 
+        self._signal_name = netatmo_home.signal_name
         self._publishers.extend(
             [
                 {
-                    "name": CLIMATE_TOPOLOGY_CLASS_NAME,
-                    SIGNAL_NAME: CLIMATE_TOPOLOGY_CLASS_NAME,
-                },
-                {
-                    "name": CLIMATE_STATE_CLASS_NAME,
-                    "home_id": self._home_id,
-                    SIGNAL_NAME: self._climate_state_class,
+                    "name": HOME,
+                    "home_id": self._home.entity_id,
+                    SIGNAL_NAME: self._signal_name,
                 },
             ]
         )
 
         self._device_name = self._home.name
-        self._attr_name = f"{MANUFACTURER} {self._device_name}"
+        self._attr_name = f"{self._device_name}"
 
         self._model: str = "NATherm1"
-        self._netatmo_type = TYPE_ENERGY
+        self._config_url = CONF_URL_ENERGY
 
         self._attr_unique_id = f"{self._home_id}-schedule-select"
 
         self._attr_current_option = getattr(self._home.get_selected_schedule(), "name")
-        self._attr_options = options
+        self._attr_options = [
+            schedule.name for schedule in self._home.schedules.values()
+        ]
 
     async def async_added_to_hass(self) -> None:
         """Entity created."""
         await super().async_added_to_hass()
 
-        for event_type in (EVENT_TYPE_SCHEDULE,):
-            self.data_handler.config_entry.async_on_unload(
-                async_dispatcher_connect(
-                    self.hass,
-                    f"signal-{DOMAIN}-webhook-{event_type}",
-                    self.handle_event,
-                )
+        self.data_handler.config_entry.async_on_unload(
```

Comment:
This really should be self.async_on_unload().

Reply:
While I understand your concert, there is no such method, neither on `NetatmoBase`, `SelectEntity` or `Entity` itself.

Reply:
https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L725